packages/frontend/src/App.tsx

```diff
@@ -1,12 +1,26 @@
 import { type Config, WagmiProvider } from "wagmi";
-import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { PersistQueryClientProvider } from "@tanstack/react-query-persist-client";
+import { createSyncStoragePersister } from "@tanstack/query-sync-storage-persister";
+import { QueryClient } from "@tanstack/react-query";
 import { createRouter, RouterProvider } from "@tanstack/react-router";
 import { getDefaultConfig, RainbowKitProvider } from "@rainbow-me/rainbowkit";
 import { MassMarketProvider } from "./MassMarketContext.ts";
 import { config } from "./wagmi.ts";
 import { routeTree } from "./routeTree.gen.ts";
 
-const queryClient = new QueryClient();
+export const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      gcTime: Infinity,
+      staleTime: Infinity,
```

Comment:
I think this is going to set the gcTime and staleTime for all queryClient globally in the app. Lets move these options to the useQuery in the keycard hook